### ISSUE TITLE
Can't edit files that were included or read by php

### ISSUE THREAD

**Issue by ElStefanos** [GetmeUK/ContentTools#568]
I mad small system that makes you to chose which page to edit using php. First i used `include 'file';` then i saw i cant edit them so i went with other method which reads the whole file and places it in webpage. Now i have yellow box around it which is indicator that is editable. But it's not. So i totally removed editable region from main page and placed it in function where php is "including" desired files. Same result.

**Comment by ElStefanos** [comment]
I would really appreciate if someone responds to this issue.

**Comment by anthonyjb** [comment]
Can you Post the HTML that's generated by your PHP application that can't be edited?

**Comment by ElStefanos** [comment]
`<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="utf-8">
    <title></title>
  </head>
  <body>
    <section>
      <h1>1</h1><h2>2</h2><h3>3</h3>
    </section>
  </body>
</html>`

This is inside the file I'm including with php.

`$content = file_get_contents('../assets/themes/default_theme/extensions/pages/html/'.$file.'.html');
echo '<div data-editable data-name=heading>'.$content.'</div>';`

  And this is a php script that loads the file.

**Comment by anthonyjb** [comment]
Sorry so the HTML has no editable regions so that's why the editor wont show, is this the HTML output by the PHP? If so why isn't it like:

```
<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="utf-8">
    <title></title>
  </head>
  <body>
    <section>
      <div data-editable data-name=heading>
          <h1>1</h1><h2>2</h2><h3>3</h3>
      </div>   
 </section>
  </body>
</html>
```

**Comment by ElStefanos** [comment]
The output from php is:
````   
      <div data-editable data-name=heading>
          <h1>1</h1><h2>2</h2><h3>3</h3>
      </div>   
```

First time i did as you in the code you provided and i didn't even had yellow indicator that it is editable.

**Comment by anthonyjb** [comment]
OK I don't know why the PHP isn't including the `<div data-editable...>` after the first time, but lets try to figure out why you can't edit the page event when it contains the editable region. I assuming you've sent me a simplified version of the page as there's no JS or CSS includes in the template, could you send through or post some where the full HTML output and the JS you have for initialising the editor.

**Comment by ElStefanos** [comment]
This is the [function](https://github.com/ElStefanos/AWBT-CMS/blob/master/build/admin/panel_sys/functions.inc.php) that loads chosen page it's on line 305.
This is the page where [content tools](https://github.com/ElStefanos/AWBT-CMS/blob/master/build/admin/page-builder.php) is.
And this is a folder where i stored [Content Tools.](https://github.com/ElStefanos/AWBT-CMS/tree/master/build/vendor/ContentTools)

**Comment by anthonyjb** [comment]
Cool ty - OK since I can't run this myself to test can I ask you do the following and let me know:

- After refreshing the page can you send me:
    - Console errors
    - An exact copy of the outputted HTML
- Can you add a `console.log(document.querySelectorAll('*[data-editable]'))` on line 7 of `editor.js` and send me the output.

If I can't work it out from this maybe we could catch up on discord or something and you can screen share so I can work through it with you?

**Comment by ElStefanos** [comment]
Thi is the output of HTML `<section class="ce-element ce-element--type-static">
      <h1>1</h1><h2>2</h2><h3>3</h3>
    </section>`
No errors on console.
And there is no output.

**Comment by anthonyjb** [comment]
OK so that looks like some how the editable region `<div data-editable data-name="heading">...</div>` included the section tag in it and so the editor interpreted that as a static element (one it can't edit) and that's why it has `class="ce-element ce-element--type-static"` against it. However, `class="ce-element ce-element--type-static"` would only be applied when you edit the page those classes are only applied when you click to edit and CT parses the page content) so I must be missing a step here. 

If you don't use PHP to drag in data dynamically and you replace `<?php PageEditorLoad($conn); ?>` with 

```
<div data-editable="" data-name="heading">
    <h1>Testing</h1>
</div>
``` 

Are you able to edit the heading in this case?

**Comment by ElStefanos** [comment]
Yes i'm able to edit after i replaced `<?php PageEditorLoad($conn); ?>`  with `<div data-editable="" data-name="heading">
    <h1>Testing</h1>
</div>` should i somehow exclude `<section>` from files im trying to load? Also am i able to edit if i have `
<div class='blah'>
    <div data-editable="" data-name="heading">
        <h1>Testing</h1>
    </div>
</div>
`

**Comment by anthonyjb** [comment]
ContentTools only handles a subset of tag/element types for editing. You can have multiple editable regions in a page and these can be any block level element pretty much, but they can only contain elements CT supports, unsupported element types will still appear but will be marked as static and you wont be able to edit them or their contents, the following is OK for example:

```
<section class="header">
    <div data-editable data-name="heading">
        <h1>Testing</h1>
    </div>
</section>
```

**Comment by ElStefanos** [comment]
I see so there is no way i can edit something that is in this format. 
```
<div data-editable="" data-name="heading">
    <div class='blah'>
        <h1>Testing</h1>
    </div>
</div>
```

**Comment by anthonyjb** [comment]
So not using that structure no. We can typically work around these restrictions (the CT site for example is all done with CT as is) but form more complex sites we use the ContentFlow library - however - this isn't documented so it's hard to get started with I'm afraid.

Sorry I couldn't be more helpful :

**Comment by ElStefanos** [comment]
Well can i implement it in my project? Do i need to have node.js or something that needs to be installed by git?

**Comment by anthonyjb** [comment]
Both ContentTools and ContentFlow are client side libraries, neither require node, any backend you want to use is fine - though it's quite a bit of work to set up support for ContentFlow currently, this is something I hope to improve in future releases.

**Comment by ElStefanos** [comment]
Okay thanks for the info i will keep my eyes on ContentFlow!
